distiller/apputils/image_classifier.py

```diff
@@ -830,30 +844,65 @@ def evaluate_model(model, criterion, test_loader, loggers, activations_collector
     if not isinstance(loggers, list):
         loggers = [loggers]
 
-    if args.quantize_eval:
-        model.cpu()
-        quantizer = quantization.PostTrainLinearQuantizer.from_args(model, args)
-        quantizer.prepare_model(distiller.get_dummy_input(input_shape=model.input_shape))
-        model.to(args.device)
+    if not args.quantize_eval:
+        return test(test_loader, model, criterion, loggers, activations_collectors, args=args)
+    else:
+        return quantize_and_test_model(test_loader, model, criterion, args, loggers,
+                                       scheduler=scheduler, save_flag=True)
+
 
-    top1, _, _ = test(test_loader, model, criterion, loggers, activations_collectors, args=args)
+def quantize_and_test_model(test_loader, model, criterion, args, loggers=None, scheduler=None, save_flag=True):
+    """Collect stats using test_loader, clone the model and quantize the clone, and finally, test it.
 
-    if args.quantize_eval:
+    args.device is allowed to differ from the model's device.
+    When args.qe_calibration is set to None, uses 0.05 instead.
+
+    scheduler - pass scheduler to store it in checkpoint
+    save_flag - defaults to save both quantization statistics and checkpoint.
+    """
+    if not (args.qe_dynamic or args.qe_stats_file):
```

Comment:
Looks good, thanks for making all the changes!

One last thing - When the `--qe-config-file <path_to_yaml>` arg is used, the contents of the YAML file are taken as-is and all other arguments are ignored. So with this if condition as-is, stats are collected but are ignored when creating the quantizer.
For the sake of merging this PR, I suggest that when a config file is used we skip stats collection. We can consider if we want to change this behavior after. So that means:
```suggestion
    if not (args.qe_dynamic or args.qe_stats_file or args.qe_config_file):
```

Actually - you created a mutually exclusive args group in the quantization args, which currently contains `qe-dynamic` and `qe-stats-file`. I think it should also include `qe-calibration` and `qe-config-file`, since these are all conflicting ways to create stats or use existing ones.

What do you say?

Reply:
see new commit